nncf/torch/graph/patterns.py

```diff
@@ -11,27 +11,131 @@
+    main_pattern.add_pattern_alternative(pattern)
 
-SINGLE_OPS = ACTIVATIONS | POOLING | N('mean') | N('layer_norm')
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    sigmoid_node = pattern.add_node(['sigmoid'])
+    mul_node = pattern.add_node(['__mul__'])
 
-ARITHMETIC = N('__iadd__') | N('__add__') | N('__mul__') | N('__rmul__')
+    pattern.add_edge(linear_ops_node, sigmoid_node)
+    pattern.add_edge(linear_ops_node, mul_node)
 
-ELTWISE_UNIFORM_OPS = BN | RELU | ACTIVATIONS
+    main_pattern.add_pattern_alternative(pattern)
 
-MATMUL = N('matmul') | N('bmm')
+    return main_pattern
+
+
+def create_linear_h_swish_act() -> GraphPattern:
+    main_pattern = GraphPattern()
+
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    bn_node = pattern.add_node(BATCH_NORMALIZATION_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+    mul_node = pattern.add_node(['__mul__'])
+
+    pattern.add_edge(linear_ops_node, bn_node)
+    pattern.add_edge(bn_node, __add__node)
+    pattern.add_edge(bn_node, mul_node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+    pattern.add_edge(__truediv__node, mul_node)
+
+    main_pattern.add_pattern_alternative(pattern)
+
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+    mul_node = pattern.add_node(['__mul__'])
+
+    pattern.add_edge(linear_ops_node, __add__node)
+    pattern.add_edge(linear_ops_node, mul_node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+    pattern.add_edge(__truediv__node, mul_node)
+
+    main_pattern.add_pattern_alternative(pattern)
+
+    return main_pattern
+
+
+def create_linear_h_sigmoid_act() -> GraphPattern:
+    pattern = GraphPattern()
+    linear_ops_node = pattern.add_node(LINEAR_OPERATIONS)
+    __add__node = pattern.add_node(['__add__'])
+    hardtanh_node = pattern.add_node(['hardtanh'])
+    __truediv__node = pattern.add_node(['__truediv__'])
+
+    pattern.add_edge(linear_ops_node, __add__node)
+    pattern.add_edge(__add__node, hardtanh_node)
+    pattern.add_edge(hardtanh_node, __truediv__node)
+
+    return pattern
+
+
+@PATTERN_GRAPH.register('graph_pattern_factory')
+class GraphPatternFactory:
+    FULL_PATTERN_GRAPH = None
+
+    @staticmethod
+    def get_graph():
+        if GraphPatternFactory.FULL_PATTERN_GRAPH is None:
+            GraphPatternFactory.FULL_PATTERN_GRAPH = GraphPatternFactory._generate_full_pattern_graph()
+        return GraphPatternFactory.FULL_PATTERN_GRAPH
+
+    @staticmethod
+    def _generate_full_pattern_graph():
```

Comment:
Instead of creating a full pattern graph here, you should create subgraphs, register them, and then create a full pattern graph by a call on the registry object. For instance:

```python
ARITHMETIC = GraphPattern(ARITHMETIC_OPERATIONS)
GRAPH_PATTERNS.register(ARITHMETIC, name='ARITHMETIC', match=False)
...
GRAPH_PATTERNS.register(ARITHMETIC + ANY_BN_ACT_COMBO, name='ARITHMETIC + ANY_BN_ACT_COMBO', match=True)

full_pattern_graph = GRAPH_PATTERNS.get_full_pattern_graph()
GRAPH_PATTERNS.visualize_all(dot_file_dump_directory)
GRAPH_PATTERNS.visualize_matched(another_dot_file_dump_directory)
full_pattern_graph.visualize_graph(yet_another_dot_file_dump_directory)
```

Reply:
Done, please, take a look